api/src/pcapi/scripts/beneficiary/import_users.py

```diff
@@ -26,23 +84,20 @@ def create_or_update_users(rows: Iterable[dict]) -> list[User]:
     users = []
     for row in rows:
         user = find_user_by_email(row["Mail"])
-        birthdate = datetime.strptime(row["Date de naissance"], "%Y-%m-%d")
         if user:
-            user.dateOfBirth = birthdate
+            user.dateOfBirth = _get_birth_date(row)
             user.setPassword(settings.STAGING_TEST_USER_PASSWORD)
         else:
-            user = users_api.create_account(
-                email=sanitize_email(row["Mail"]),
-                password=settings.STAGING_TEST_USER_PASSWORD,
-                birthdate=birthdate,
-                is_email_validated=True,
-                send_activation_mail=False,
-                remote_updates=False,
-            )
-            deposit = payments_api.create_deposit(user, "import_users (csv)")
-            repository.save(deposit)
+            try:
+                role = UserRole(row["Role"])
+            except ValueError:
+                role = None
```

Comment:
Est-ce qu'on ne voudrait pas échouer bruyamment en cas de typo dans le fichier ? Par exemple, si je laisse un espace dans "PRO " dans le fichier CSV (une typo qui pourrait passer inaperçu à la relecture), le script actuel créerait un bénéficiaire. Je propose de laisser le code lever une erreur, ce sera plus explicite.

```suggestion
            role = UserRole(row["Role"])
```

Reply:
La colonne `Role` peut être vide si on veut créer un utilisateur sans qu'il devienne bénéficiaire. Dans ce cas, `UserRole("")` renverra une exception. Mais je peux faire sans `try except` :
`role = UserRole(row["Role"]) if row["Role"] else None`

Reply:
@prouzet : y a -t-il un cas d'utilisation pour lequel on voudrait créer un utilisateur sans rôle ?